lib/matplotlib/patches.py

```diff
@@ -806,6 +806,26 @@ def get_xy(self):
         """Return the left and bottom coords of the rectangle as a tuple."""
         return self._x0, self._y0
 
+    def get_edge_midpoints(self):
+        """
+        Return midpoints of the edges of the rectangle, moving anti-clockwise
+        from the center of the left-hand edge.
+        """
+        return self.get_patch_transform().transform(
+            [(0, 0.5), (0.5, 1), (1, 0.5), (0.5, 0)])
+
+    def get_corners(self):
+        """
+        Return the corners of the rectangle, moving anti-clockwise from
```

Comment:
This one is clockwise, now!

Reply:
Should be fixed

Reply:
Um, no?

``` 
(0, 1)-----(1, 1)
   |          |
(0, 0)-----(1, 0)
```
So the order should be `[(0, 0), (1, 0), (1, 1), (0, 1)] `.

Reply:
ahhhh this is driving me mad! Thanks for the diagram...